gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnService.java

```diff
@@ -462,7 +448,7 @@ public synchronized void requestTargetNumberOfContainers(YarnContainerRequestBun
       for(; requestedContainerCount < desiredContainerCount; requestedContainerCount++) {
         requestContainer(Optional.absent(), yarnContainerRequestBundle.getHelixTagResourceMap().get(currentHelixTag));
       }
-      requestedContainerCountMap.put(currentHelixTag, requestedContainerCount);
+      requestedContainerCountMap.put(currentHelixTag, desiredContainerCount);
```

Comment:
I think we discussed this as part of PR#3487: why would desiredContainerCount and requestedContainerCount not be the same here?

Reply:
I didn't put this case into consideration: in the desiredContainerCount is smaller than requestedContainerCount, the for-loop wouldn't update the requestedContainerCount. So need to use desiredContainerCount instead.